syntax = "proto3";

package data_transfer_api;

service KeyValueService {
  rpc GetValue(GetValueRequest) returns (GetValueResponse) {}
  rpc StoreValue(StoreValueRequest) returns (StoreValueResponse) {}
}

message Value {
  bytes payload = 1;
}

message StoreValueResponse {
  int32 code = 1;
  string message = 2;
}

message GetValueRequest {
  string key = 1;
}

message GetValueResponse {
  bool key_found = 1;
  optional Value value = 2;
}

message StoreValueRequest {
  string key = 1;
  Value value = 2;
}
